TicTacToe/main.cpp

```diff
@@ -0,0 +1,79 @@
+#include <iostream>
+#include <queue>
+#include <string>
+#include <cassert>
+#include "PlayField.h"
+#include "TreeNode.h"
+
+struct Counter {
+    int crossesWin = 0;
+    int noughtsWin = 0;
+    int draw = 0;
+
+    Counter operator+(Counter elem) {
+        Counter res = *this;
+        res.crossesWin += elem.crossesWin;
+        res.noughtsWin += elem.noughtsWin;
+        res.draw += elem.draw;
+        return res;
+    }
+
+     friend std::ostream& operator<<(std::ostream &out, Counter &c)  {
+         out << "X " << c.crossesWin << ", 0 " << c.noughtsWin << ", D " << c.draw << std::endl;
+         return out;
+    }
+
+    int overall(){
+        return crossesWin + noughtsWin + draw;
+    }
+};
+
+void BuildSubTree(TreeNode& node) {
+    if(node.value().checkFieldStatus() == PlayField::fsNormal)
+        for(auto item : node.value().getEmptyCells()) {
+            node.addChild(node.value().makeMove(item));
+            BuildSubTree(node[node.childCount()-1]);
+        }
+}
+
+void WalkTree(const TreeNode& node) {
+    auto totalCount = Counter();
+    for(int i = 0; i < node.childCount(); ++i) {
+        auto localCount = Counter();
+        std::queue<TreeNode> queue;
+        queue.push(node[i]);
+        while(!queue.empty()) {
+            auto active = queue.front();
+            switch (active.value().checkFieldStatus()) {
+                case PlayField::fsCrossesWin:
+                    localCount.crossesWin++;
+                    break;
+                case PlayField::fsNoughtsWin:
+                    localCount.noughtsWin++;
+                    break;
+                case PlayField::fsDraw:
+                    localCount.draw++;
+                    break;
+                case PlayField::fsInvalid:
+                    assert(active.value().checkFieldStatus() == PlayField::fsInvalid);
+                    break;
+            }
+            for (int j = 0; j < active.childCount(); ++j)
+                queue.push(active[j]);
+            queue.pop();
+        }
+        node[i].value().print();
+        std::cout << localCount;
+        totalCount = totalCount + localCount;
+    }
+    std::cout <<"TOTAL " << totalCount;
+    std::cout << "OVERALL " << totalCount.overall() << std::endl;
+}
+
+int main() {
+    PlayField pf0;
+    TreeNode node0(pf0);
```

Comment:
```suggestion
    TreeNode node0(PlayField());
```
Всё равно pf0; не используется...

Reply:
(PlayField())